codecarbon/core/emissions.py

```diff
@@ -78,6 +79,11 @@ def get_private_infra_emissions(self, energy: Energy, geo: GeoMetadata) -> float
         :param geo: Country and region metadata
         :return: CO2 emissions in kg
         """
+        if co2_signal.is_available():
+            try:
+                return co2_signal.get_emissions(energy, geo)
+            except co2_signal.CO2SignalAPIError as e:
```

Comment:
catch any exception

Reply:
Requests itself can raises `requests.exceptions.Exception` in various situations and the parsing of the response body can raises various exceptions, let's catch `except Exception` here